app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ingredients_analysis/IngredientsAnalysisProductFragment.java

```diff
@@ -0,0 +1,103 @@
+package openfoodfacts.github.scrachx.openfood.views.product.ingredients_analysis;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+
+import java.util.ArrayList;
+
+
+import butterknife.BindView;
+import openfoodfacts.github.scrachx.openfood.R;
+import openfoodfacts.github.scrachx.openfood.fragments.BaseFragment;
+import openfoodfacts.github.scrachx.openfood.models.Ingredient;
+import openfoodfacts.github.scrachx.openfood.models.Product;
+import openfoodfacts.github.scrachx.openfood.models.State;
+import openfoodfacts.github.scrachx.openfood.network.OpenFoodAPIClient;
+import openfoodfacts.github.scrachx.openfood.views.product.ingredients_analysis.adapter.IngredientAnalysisRecyclerAdapter;
+import retrofit2.Response;
+
+public class IngredientsAnalysisProductFragment extends BaseFragment {
+    @BindView(R.id.ingredient_analysis_recycler_view)
+    RecyclerView mRecyclerView;
+
+    private OpenFoodAPIClient api;
+    private Product product;
+    private ArrayList<Ingredient> ingredients;
+    private IngredientAnalysisRecyclerAdapter adapter;
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        Intent intent = getActivity().getIntent();
+        State state = (State) intent.getExtras().getSerializable("state");
+        product = state.getProduct();
+        api = new OpenFoodAPIClient(getActivity());
+
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState)
+    {
+        mRecyclerView=new RecyclerView(getContext());
+        ingredients=new ArrayList<>();
+        Callback<JsonNode> callback=new Callback<JsonNode>() {
+            @Override
+            public void onResponse(@NonNull Call<JsonNode> call, Response<JsonNode> response) {
```

Comment:
`onResponse` and `onFailure` logic must be moved to app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java file. Refer to this file to see how to do it.

Reply:
I've done all the requested changes :-) please review.